### ISSUE TITLE
Request: Long range scanner.

### ISSUE THREAD

**Issue by DutchTrader** [endless-sky/endless-sky#4712]
**Is your feature request related to a problem? Please describe.**
This is more related to me finding it gamey to reload after getting shot down or when a mission fails. This makes tactics and flying skills more important. However, one unlucky jump into the next system can ruin the game and cause a restart from scratch. While this becomes less and less, these days it can still happen with a flimsy craft (like the startup shuttle), or a mission involving a flimsy or disabled ship.

**Related Issue Links**
A quick Google search turned up blank.

**Describe the solution you'd like**
A nice to have would be a long-range scanner, that can take a sneak peak into the next system you're jumping into. A simple radar view would already do a lot of good. A list of ship names, types and affiliations would make it perfect. Would that be possible?

**Describe alternatives you've considered**
None.

**Additional context**
N/A

**Comment by Amazinite** [comment]
There was actually a discussion about this on the Discord very recently. I'd past quotes from that discussion, but the search feature on Discord seems to be broken right now, so I'll post some of that later.

I'd suggest looking into #3572, which is somewhat similar in how it tells you the danger to expect in a system as opposed to what exactly is in the system. This idea was also discussed in relation to #362 and lowering the effects of fleet RNG on gameplay.

**Comment by Amazinite** [comment]
>Zitchas:
>https://www.reddit.com/r/endlesssky/comments/e6213a/the_best_and_worst_things_about_this_game/
>Just a nother discussion about how much the RNG is problematic, and how people dislike having "reload from the last save" being a standard and expected part of the mechanic.
>I'm inclined to agree with that part: The game difficulty should not be based around the assumption that most players will die multiple times before successfully completing a combat mission.

>Me:
>I get that the RNG can be annoying, but if we have no system to replace it with then we can't change it.
>A change that we could make now: https://github.com/endless-sky/endless-sky/issues/4039 While this doesn't remove the RNG aspect, it does lessen its effects by making fleets less common.
>A change that is planned that would be a solution by removing the immediate fail: escape pods.
>A possible change that we could look into: Spawn fleets (or queue fleets to be spawned) in the surrounding systems instead of just in the current system, then give the player some way of knowing the ships in adjacent systems. That way they could see "Hey, there's 2 small pirate fleets in this system and nothing else. I should go around."

>Zitchas:
>The escape pods would be great. Likewise, the possibility of being captured and released on a nearby world would be interesting. It'd be very interesting if pirates would disable, board, loot, and then repair the player - with the caveat that they auto-bribe after looting. Basically a form of "We just took your money, we want you to be able to make more money so we can loot you again in the future" sort of thing.
>Actually, being able to see what ships are in adjacent systems would be a GREAT additional "feature" to build into the tactical scanner. It fits the description, and would transform it from a "I have this because I like having more stuff in my GUI" thing into a "This lets me tactically plan my routes!" sort of thing. Much more useful.
 
>Me:
>For the third idea, I can imagine having some sort of "hyperlink warning network", possibly in the form of an outfit that lets you access it.
>Then we just hand wave once you get into alien space.

A potential image from @beccabunny.
![image](https://user-images.githubusercontent.com/17688683/70417222-c7bb9a80-1a2e-11ea-81dc-ab6662510ff0.png)

**Comment by Amazinite** [comment]
UI wise, I imagine that having such a scanner installed would bring up an additional panel on the map when you have a neighboring system selected. It would only list the number of ships per government. If there are too many governments in the system, then the last line would say "X other ships". If you have a system selected that isn't a neighboring system, it'd say "No link to this system" or something to that effect.
![Screenshot_2](https://user-images.githubusercontent.com/17688683/70418211-ffc3dd00-1a30-11ea-8427-dfa2f12ab32d.png)

**Comment by tehhowch** [comment]
Since there are no ships in a neighboring system except those that recently jumped from the player's system or the player's escorts, this isn't really something that can be done at the moment. Without first adding in some NPC "linger" behaviors, it doesnt make sense to spawn ships in adjacent systems as they would immediately jump away or land, or be cleaned up by the game after about 20 seconds. 

In short: not possible without mechanics changes, and then an enhancement.

Additionally, from a gameplay perspective, how does the player know what ships will be in a system a day into the future?

**Comment by beccabunny** [comment]
>Additionally, from a gameplay perspective, how does the player know what ships will be in a system a day into the future?

Well, gameplay wise, it would be the sum of the ships warping there and the ships departing from the eventual planets in the system

**Comment by Zitchas** [comment]
From a lore perspective, I agree with BeccaBunny. We just need to change the description away from a "this is a scan of what's in the system" description, to a "this is what's docked, landed, and in-transit to the system and thus a prediction of what will be there" sort of description.

I think it might make it more interesting (and fit better with the "predictions" concept) if the numbers were not provided as exact figures. Instead using a table along the lines of what was used in the Heroes of Might and Magic games:

Few | 1-4
Several | 5-9
Pack | 10-19
Lots | 20-49
Horde | 50-99
Throng | 100-249
Swarm | 250-499
Zounds | 500-999
Legion | 1000+

Of course we'll want to modify the ranges and terms a bit, both to avoid stepping on HOMM's toes and to get something that fits our specific circumstances a bit better. Probably tighten up the lower scales a bit, since we'll want to have more variation in the smaller numbers as that's what is usually relevant for us. 

For changing effects based on power/effectiveness (and thus giving the option to have different types of this for different tiers), three things come to mind: Range, Precision, and whether or not it is restricted to following hyperjump routes. Likewise, fall-off would be interesting attribute to have as well. This would be the range where the outfit's precision drops from whatever it is normally down to a minimum resolution. So within that range it gives normal results, beyond that range it only lists what ship factions are present. No numbers at all. 
This would give us a number of options for progression over time:

A) low precision, 1 jump range, limited to hyperjump routes. Either default available outfit or easy/common unlock via a short mission chain from an outfit manufacturer.
B) low precision with falloff beyond 1 jump, 2 jump range, limited to hyperjump routes. Available as a mid campaign unlock during the human campaign.
C) low precision, 1 jumpdrive range, is not limited to jump routes. Found on low-tier Pug and possibly Korath ships. Possible late human campaign unlock (sometime after taking down giftbringer, based on the hyperspace tracking tech)
4) medium precision, 1 jump range, limited to following hd routes. Found on wanderer, heliarch, and korath automata ships
5) medium precision, 2 jump range, not limited. Found on Quarg, and high-tier pug ships.
6) high precision (exact numbers), 2 jump range, not limited. Found on Drak and unseen-as-of-yet drak equivalent pug ships.
